KarhooSDK/Service/Loyalty/LoyaltyEarn/KarhooLoyaltyEarnInteractor.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  KarhooLoyaltyEarnInteractor.swift
+//  KarhooSDK
+//
+//  Created by Edward Wilkins on 17/11/2021.
+//  Copyright © 2021 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class KarhooLoyaltyEarnInteractor: LoyaltyEarnInteractor {
+    
+    private let requestSender: RequestSender
+    private var identifier: String?
+    private var currency: String?
+    private var amount: Int?
+    private var points: Int?
+    
+    init(requestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared)) {
+        self.requestSender = requestSender
+    }
+    
+    func set(identifier: String, currency: String, amount: Int, points: Int) {
+        self.identifier = identifier
+        self.currency = currency
+        self.amount = amount
+        self.points = points
+    }
+    
+    func execute<T>(callback: @escaping CallbackClosure<T>) where T : KarhooCodableModel {
+        requestSender.requestAndDecode(payload: nil,
```

Comment:
we better to have gaurd let for identifier and currency and instead of `?? 0` you can have amount: Int? = 0

Reply:
Sure can do

Reply:
We need the guard to call the callback with the appropriate error (in this case missing / invalid currency) before returning, otherwise the caller will be left hanging with no clue as to what has happened to their call.